### ISSUE TITLE
Error installing package

### ISSUE THREAD

**Issue by jayanthnd** [diegomura/react-pdf#337]
**OS: Mac OS High Sierra**

**React-pdf version: 1.0.0-alpha.18**

**Description: Unable to yarn/npm install @react-pdf/renderer**

**How to replicate issue including code snippet (if applies): Tried running `yarn install @react-pdf/renderer` and get the following error:
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
make: *** [Release/obj.target/nbind/sources/yoga/Utils.o] Error 1
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:262:23)
gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:240:12)
gyp ERR! System Darwin 17.6.0
gyp ERR! command "/usr/local/Cellar/node/10.11.0/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "configure" "build"
gyp ERR! cwd /Users/node_modules/yoga-layout
gyp ERR! node -v v10.11.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! yoga-layout@1.9.3 build:node: `npm run copy-sources && autogypi && node-gyp configure build`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the yoga-layout@1.9.3 build:node script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
**

_You can make use of [react-pdf REPL](https://react-pdf/repl) to share the snippet_

I've seen a couple of other threads with similar issues, just wondering if there's a workaround for it.

**Comment by diegomura** [comment]
That’s an issue about Yoga installation on your system. I would reccomend you to look for more information in the yoga-layout repo, or the ones that were created here (such as #327)

I’m not available right now and I’ll try to take a closer look when I got back from vacations

**Comment by jayanthnd** [comment]
Thanks for the reply @diegomura. I fixed the issue, turned out it had to do with how the header files for the yoga library were getting referenced in my local setup.

I do have a follow-up question though - I'm using react 15.4, is this library compatible with it?
Also, I have the following implementation - 
`import React, {Component} from 'react';
import { PDFDownloadLink, BlobProvider, Page, Text, View, Document, StyleSheet } from '@react-pdf/renderer';

// Create styles
const styles = StyleSheet.create({
  page: {
    flexDirection: 'row',
    backgroundColor: '#E4E4E4'
  },
  section: {
    margin: 10,
    padding: 10,
    flexGrow: 1
  }
});

// Create Document Component
const MyDocument = () => (
  <Document>
    <Page size="A4" style={styles.page}>
      <View style={styles.section}>
        <Text>Section #1</Text>
      </View>
      <View style={styles.section}>
        <Text>Section #2</Text>
      </View>
    </Page>
  </Document>
);

export default class PDFDocument extends Component {
  render() {
    return (
      <div>
        <PDFDownloadLink document={MyDocument} fileName="example.pdf">
          Download
        </PDFDownloadLink>
      </div>
    );
  }
}
`

And I import the component as `import PDFDocument from "../PDF";` in a different file. And I get the following error - 
`Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.`

Any thoughts?

**Comment by jayanthnd** [comment]
[UPDATE] I removed the `PDFDownloadLink` wrapper and it does not throw that error anymore. Also noticed that my interpreter is unable to resolve PDFDownloadLink import.

**Comment by jayanthnd** [comment]
[UPDATE] Based on https://github.com/diegomura/react-pdf/issues/317, I changed my code to look like this but I run into the same error:
```
import React, {Component} from 'react';
import { PDFDownloadLink, Page, Text, View, Document, StyleSheet } from '@react-pdf/renderer/dist/react-pdf.browser.es.js';
// import { Page, Text, View, Document, StyleSheet } from '@react-pdf/renderer';

// Create styles
const styles = StyleSheet.create({
  page: {
    flexDirection: 'row',
    backgroundColor: '#E4E4E4'
  },
  section: {
    margin: 10,
    padding: 10,
    flexGrow: 1
  }
});

// Create Document Component
const MyDocument = () => (
  <Document>
    <Page size="A4" style={styles.page}>
      <View style={styles.section}>
        <Text>Section #1</Text>
      </View>
      <View style={styles.section}>
        <Text>Section #2</Text>
      </View>
    </Page>
  </Document>
);

export const PDFDocument = () => (
  <div>
    <PDFDownloadLink document={MyDocument} fileName="example.pdf">
      Download
    </PDFDownloadLink>
  </div>
);
```

Error:
```
Uncaught Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.
```

**Comment by luskin** [comment]
@jayanthnd can you please expand on how you fixed this problem? We can't install the package.

**Comment by rogierslag** [comment]
```
xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance
```

It suggest you need to install xcode on OSX, and accept its license agreement as well prior to installing

**Comment by jorgenhovind** [comment]
@luskin, I got the same error after upgrading to macOS Mojave. Run `xcode-select --install` to fix it. [https://apple.stackexchange.com/questions/254380/macos-mojave-invalid-active-developer-path](url)

**Comment by luskin** [comment]
@rogierslag @jorgenhovind Thanks for the responses but I have previously tried all of those solutions and none worked (I am also an iOS developer so my Xcode is up-to-date and fully installed). Turns out react-pdf does not work with certain versions of node. I fully upgraded everything to `9.10.0` and that eventually solved the problem. I'm not sure for others, but for me anything sub `8.9.3` wouldn't install.

**Comment by elrumordelaluz** [comment]
same _issue_ with `youga-layout` in _CentOS_

**Comment by diegomura** [comment]
Hey. Thanks for reporting this. 

I'm aware of this issue. Unfortunately there's no much I can do, since it's a Yoga related issue and it's a core dependency of this package. 

What might solve it can be using [yoga-dom](https://github.com/vincentriemer/yoga-dom), which is the same lib but built already for webassembly and not asm as `yoga-layout`. I already tried to add it to the project, but unfortunately there's a simple but yet destructive problem: `yoga-dom` exports a promise, and not the yoga objects directly. I couldn't deal with it without changing the react-pdf API (which I was trying to avoid). Feel free to someone to tackle this. Help would be very welcomed

**Comment by sl-ulken** [comment]
@jayanthnd I got: 
```console
Warning: Functions are not valid as a React child. This may happen if you return a **Component** instead of **<Component />** from render. Or maybe you meant to call this function rather than return it.
```

Doing as suggested in warning message got it working for me. I.e. changing `document={MyDocument}` to `document={<MyDocument />}` in your case.

**Comment by diegomura** [comment]
Should be fixed on #364
